### DISCUSSION TITLE
Objetos em javascript

### DISCUSSION THREAD

**Discussion by brunoravanhani** [frontendbr/forum#107] [Perguntas e Respostas]
Quando digo objetos quero dizer como uma classe Usuario em Java com seus gets e sets que serve apenas para transportar dados vindos de um banco de dados relacional. Como vocês trabalham com isso em javascript?

Não sei se me fiz entender... é que trabalho numa empresa que desenvolve soluções corporativas mobile e estamos migrando do android pro híbrido então há aplicações com várias tabelas relacionais que precisam de tratamento na ponta e nisso me surgiu uma duvida de quais padrões utilizar.

**Comment by icaromh** [comment]
@brunoravanhani Dá uma olhada nesse repositório aqui:

https://github.com/getify/You-Dont-Know-JS/blob/master/this%20&%20object%20prototypes/README.md#you-dont-know-js-this--object-prototypes

Edit: Acabei que não respondi nada pra ti hahah. Nunca fiz uso de classes (ES6), então não posso te dar uma força nisso. Quanto a objetos propriamente dito, depende do que to usando. Se estou usando angular, uso o guideline deles. Com Backbone, aqui na empresa, trabalhamos com as Views e tal. Há tempos que não crio um prototype :|

**Comment by vitorabner** [comment]
@brunoravanhani Não sei se entendi muito bem sua pergunta. 

O JS ficará no client consumindo APIs do seu server que estará em Java, e ele terá acesso ao seu banco de dados relacional. 

Se você está usando Ionic(AngularJS), é só criar um Service que irá retornar um novo objeto Usuário baseado no dados retornados por sua API(Java). 

Faz muito tempo que não trabalho com Java, porém essa sua classe Usuário(que é uma POJO) servirá de ponta para sua API.

Esse repositório que o @icaromh  mostrou é top demais! Recomendo.

**Comment by brunoravanhani** [comment]
Então @vitorabner, ficou mesmo confusa a pergunta, mas o que preciso é de algo que se assemelhe ao POJO, porém o utilizando javascript. Eu dei o exemplo do java porque é o que mais trabalhei, mas o que vai acontecer é que eu receberei os dados da API e salvo no banco utilizando um [plugin](https://github.com/litehelpers/Cordova-sqlite-storage) que simula o SQLite, então eu vou acessar um "banco relacional" com o javascript e gostaria de ter um bom controle sobre os relacionamentos e tals, pois existem aplicações com umas 30 tabelas. E assim, eu não estou utilizando Ionic, por uma decisão no começo do projeto (que eu me arrependo agora rsrsrs).

@icaromh valeu, eu acho que esse repositório vai ajudar bastante, principalmente agora que estou estudando javascript de verdade e da forma certa.

**Comment by vitorabner** [comment]
Ainda não tive experiência com um caso desse, mas já trabalhei com NodeJS + MySQL + SequelizeJS(ORM). O SequelizeJS(http://docs.sequelizejs.com/en/latest/) trabalha justamente nessa parte, mas foi no backend.

Eu encontrei esse persistencejs(https://github.com/coresmart/persistencejs) que faz esse mapeamento no front. Vê se ajuda. (Edit: parece que ta abandonado :/).

**Comment by lagden** [comment]
@brunoravanhani é tudo igual!! apenas a sintaxe que muda um pouco.
Uma das soluções é trabalhar com `API REST` e com response em `JSON` para facilitar a vida do front.
Você pode continuar utilizando seu `Java` como backend.
Mas se quiser novas experiências, existe o [Lovefield](https://github.com/google/lovefield)
Um outro que eu gosto bastante é o [CouchDB](http://couchdb.apache.org/) + [PouchDB](https://pouchdb.com/)
Meteor para desenvolver app híbrido, etc... E não acaba mais!!!
